portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -133,7 +149,53 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     commentEntity.setProperty("text", request.getParameter("comment"));
     commentEntity.setProperty("date", DateFor.format(new Date()));
     commentEntity.setProperty("timestamp", System.currentTimeMillis());
+    // Get the URL of the image that the user uploaded to Blobstore.
+    String imageUrl = getUploadedFileUrl(request, response, "image");
+    if (imageUrl != null) {
+      commentEntity.setProperty("imageURL", imageUrl);
+    }
     datastore.put(commentEntity);
-    response.sendRedirect("index.html");
+    response.sendRedirect("/index.html");
+  }
+
+  /** Returns a URL that points to the uploaded file, or null if the user didn't upload a file. */
+  private String getUploadedFileUrl(HttpServletRequest request, HttpServletResponse response, String formInputElementName) throws IOException {
+    BlobstoreService blobstoreService = BlobstoreServiceFactory.getBlobstoreService();
+    Map<String, List<BlobKey>> blobs = blobstoreService.getUploads(request);
+    List<BlobKey> blobKeys = blobs.get(formInputElementName);
+
+    // User submitted form without selecting a file, so we can't get a URL. (dev server)
+    if (blobKeys == null || blobKeys.isEmpty()) {
+      return null;
+    }
+
+    // Our form only contains a single file input, so get the first index.
+    BlobKey blobKey = blobKeys.get(0);
+
+    // User submitted form without selecting a file, so we can't get a URL. (live server)
+    BlobInfo blobInfo = new BlobInfoFactory().loadBlobInfo(blobKey);
+    if (blobInfo.getSize() == 0) {
+      blobstoreService.delete(blobKey);
+      return null;
+    }
+
+    // Check the validity of the file here, to make sure it's an image file.
+    if (blobInfo.getContentType().contains("image") == false) {
+      response.sendError(HttpServletResponse.SC_BAD_REQUEST, "Only images allowed!");
+      blobstoreService.delete(blobKey);
+    }
+
+    // Use ImagesService to get a URL that points to the uploaded file.
+    ImagesService imagesService = ImagesServiceFactory.getImagesService();
+    ServingUrlOptions options = ServingUrlOptions.Builder.withBlobKey(blobKey);
+
+    // To support running in Google Cloud Shell with AppEngine's devserver, we must use the relative
+    // path to the image, rather than the path returned by imagesService which contains a host.
```

Comment:
This sounds weird. Is it clear to you why that would be the case?

Reply:
Yes, we only need the relative path of the image, for it to be added(concatenated) in the url of my own page, I don't need the whole url in which the image is actually stored in the Blobstore service.